lib/puppetserver/ca/utils/http_client.rb

```diff
@@ -136,10 +137,16 @@ def delete(url_override = nil, headers = {})
         # Like URI, but not... maybe of suspicious value
         URL = Struct.new(:protocol, :host, :port,
                          :endpoint, :version,
-                         :resource_type, :resource_name) do
+                         :resource_type, :resource_name, :query) do
                 def full_url
-                  protocol + '://' + host + ':' + port + '/' +
-                  [endpoint, version, resource_type, resource_name].join('/')
+                  if query.empty?
+                    return  protocol + '://' + host + ':' + port + '/' +
+                         [endpoint, version, resource_type, resource_name].join('/')
+                  else
+                    return protocol + '://' + host + ':' + port + '/' +
+                         [endpoint, version, resource_type, resource_name].join('/') +
+                         "?" + URI.encode_www_form(query)
```

Comment:
Can we conditionally add the '?' and the params only if params isn't empty?

Reply:
Isn't the `if query.empty?` already check for whether params aren't empty in this if/else block?

Reply:
yeah, I think it does..
Thought this could be simplified to put the base url into a variable and then conditionally append the query part if it's not empty.

Reply:
yes, what molly said is what I meant, sorry.